### PR TITLE
Run helm update automatically when upgrading NE

### PR DESCRIPTION
**Pull request by giovannibianco**
Here we add a call to run "helm repo update" automatically when upgrading a Nuvlaedge.
If this is not done then newer versions are impossible to upgrade to...
This should be merged sooner than later as the upgrade will not work to newer versions that have been deployed since the first deployment of a NuvlaEdge.

### CODE REVIEW

File: job_engine/connector/kubernetes.py [hunk 1]
```diff
@@ -733,6 +733,48 @@ def reboot(self):
...
+        """
+        log.debug("Calling for components...\n")
+        resources = \
+            self.api.search('nuvlabox-status', \
+                filter=f"parent=\"{self.nuvlabox_id}\" ", \
+                ).resources
+        for resource in resources:
+            nuvlabox_status_string = str(resource.id)
+            log.debug(f"Nuvlabox status string: {nuvlabox_status_string}")
+            nb_status_data = self.api.get(nuvlabox_status_string).data
```
Comment (konstan) [inline job_engine/connector/kubernetes.py:749]: If on Nuvla then use select.

File: job_engine/connector/kubernetes.py [hunk 2]
```diff
@@ -746,9 +788,18 @@ def update_nuvlabox_engine(self, **kwargs):
...
+            (f"The components from nuvlabox_status:\n{installed_components}")
+        if not installed_components:
+            result = "The installed components cannot be found. Please try in a few minutes"
+            log.info(result)
+            return result, 98
+
+        project_name = self.get_helm_name()
+        # result, return_code = self.helm_check_multiple(project_name)
+        # if return_code != 0:
+          #   return result, return_code
```
Comment (konstan) [inline job_engine/connector/kubernetes.py:802]: Remove comments

File: job_engine/connector/kubernetes.py [hunk 3]
```diff
@@ -733,6 +733,48 @@ def reboot(self):
             log.debug(f'The result of the reboot action: {reboot_result}')
         return reboot_result
 
+    @property
```
Comment (konstan) [inline job_engine/connector/kubernetes.py:736]: I wouldn't define this method as a property as it's rather "heavy". Please remove the decorator and make an explicit call to it where it is used.

Regarding the name. I would suggest deployed_ne_components().

File: job_engine/connector/kubernetes.py [hunk 4]
```diff
@@ -733,6 +733,48 @@ def reboot(self):
...
+        log.debug("Calling for components...\n")
+        resources = \
+            self.api.search('nuvlabox-status', \
+                filter=f"parent=\"{self.nuvlabox_id}\" ", \
+                ).resources
+        for resource in resources:
+            nuvlabox_status_string = str(resource.id)
+            log.debug(f"Nuvlabox status string: {nuvlabox_status_string}")
+            nb_status_data = self.api.get(nuvlabox_status_string).data
+            log.debug(f"The nuvlabox-status data is:\n{json.dumps(nb_status_data, indent=2)}")
```
Comment (konstan) [inline job_engine/connector/kubernetes.py:750]: I would suggest not using f-string and provide json serialised string as param. Also, the json serialisation is heavy, so hiding everything behind the log level check would be better.              
```python
if log.level == logging.DEBUG:
   log.debug("The nuvlabox-status data is:\n %s",
                       json.dumps(nb_status_data, indent=2))
```

File: job_engine/connector/kubernetes.py [hunk 5]
```diff
@@ -775,6 +829,33 @@ def update_nuvlabox_engine(self, **kwargs):
...
+    def helm_update_the_repo(self):
+
+        helm_repo_update_job_name = self.create_job_name("helm-repo-update")
+        helm_repo_update_cmd="'helm repo update'"
+        helm_repo_update_result = \
+            self.run_helm_container(helm_repo_update_job_name, helm_repo_update_cmd)
+        log.info(f"Helm update result:\n {helm_repo_update_result}")
+
+    def get_helm_name(self):
+        """text"""
```
Comment (konstan) [inline job_engine/connector/kubernetes.py:841]: Provide a better documentation or remove it.

**Review by konstan — CHANGES_REQUESTED** [review]
Please check comments.

File: job_engine/connector/kubernetes.py (hunk 3 above)
Comment (giovannibianco) [reply job_engine/connector/kubernetes.py:736]: Changed to an explicit call and changed the name for clarity

File: job_engine/connector/kubernetes.py (hunk 4 above)
Comment (giovannibianco) [reply job_engine/connector/kubernetes.py:750]: I have changed this at this point and also above where JSON is dumped.

File: job_engine/connector/kubernetes.py [hunk 6]
```diff
@@ -733,6 +733,48 @@ def reboot(self):
...
+    @property
+    def deployed_components(self):
+        """
+        Function to get the installed components
+        """
+        log.debug("Calling for components...\n")
+        resources = \
+            self.api.search('nuvlabox-status', \
+                filter=f"parent=\"{self.nuvlabox_id}\" ", \
+                ).resources
```
[the remark this answers is not in the document]
Comment (giovannibianco) [reply job_engine/connector/kubernetes.py:755]: I have added a check on

if os.path.exists(f"{NUVLAEDGE_SHARED_PATH}/.nuvlabox_status"):

which then takes the components from there if it exists and returns the values.

**Review by konstan — APPROVED** [review]

**Pull request merged**